middleware.ts

```diff
@@ -26,6 +26,10 @@ export async function middleware(req: NextRequest) {
   if (process.env.NODE_ENV !== "production") {
     return NextResponse.next();
   }
+  // We only run blue-green deployments when accesing from production urls
+  if (req.nextUrl.hostname !== process.env.VERCEL_PROJECT_PRODUCTION_URL) {
```

Comment:
This might be fine for the demo, but other folks would need to do something more sophisticated if they have multiple production URLs.

You can do `if (req.nextUrl.hostname === process.env.VERCEL_URL) return …`